### ISSUE TITLE
Checkbox form example incorrect

### ISSUE THREAD

**Issue by puradox** [huntabyte/shadcn-svelte#948]
### Describe the bug

All forms using the `Checkbox` component in the same manner as the [doc example](https://www.shadcn-svelte.com/docs/components/checkbox#form) will omit its `value`, leading to the FormData always reporting this field as `false`.

This occurs because `<CheckboxPrimitive.Input />` is not included in the `Checkbox` component. Thus any forms will not include the hidden `<input>` element nor will it include its `value` attribute, which is the only attribute processed by forms.

Note: the `checked` attribute is **not** sufficient for forms.

See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#additional_attributes

### Reproduction

Submit a form using the [doc example](https://www.shadcn-svelte.com/docs/components/checkbox#form) and inspect the form data sent to the server.

The fix is to add `CheckboxPrimitive.Input` to `Checkbox`

```
<CheckboxPrimitive.Root ...>
	<CheckboxPrimitive.Indicator ...>
		...
	</CheckboxPrimitive.Indicator>
	<CheckboxPrimitive.Input />
</CheckboxPrimitive.Root>
```

### System Info

```bash
System:
    OS: Windows 11 10.0.22621
    CPU: (32) x64 AMD Ryzen 9 5950X 16-Core Processor
    Memory: 15.69 GB / 31.92 GB
  Binaries:
    Node: 21.4.0 - C:\Program Files\nodejs\node.EXE
    npm: 10.2.4 - C:\Program Files\nodejs\npm.CMD
    pnpm: 8.11.0 - ~\AppData\Local\pnpm\pnpm.EXE
    bun: 1.0.18 - ~\.bun\bin\bun.EXE
  Browsers:
    Edge: Chromium (122.0.2365.92)
    Internet Explorer: 11.0.22621.1
  npmPackages:
    @sveltejs/kit: ^2.5.4 => 2.5.4 
    bits-ui: ^0.11.8 => 0.11.8 
    formsnap: ^0.5.1 => 0.5.1 
    lucide-svelte: ^0.298.0 => 0.298.0 
    svelte: ^4.2.12 => 4.2.12 
    sveltekit-superforms: ^2.11.0 => 2.11.0
```

### Severity

blocking an upgrade

**Comment by puradox** [comment]
The doc example actually has a hidden input that I missed when migrating.
```
<input name={attrs.name} value={$formData.mobile} hidden />
```
Closing the bug.

**Comment by memestageceo** [comment]
The Docs still aren't clear on this issue. Though the example contains the hidden input field, it isn't clear that it's required to actually pass the value binded to the checkbox when the form is sumitted.

It requires further improvement and clear instruction on a required hidden input.

The only reason I was able to figure it out is because I went through the docs of Bits UI which had a <Checkbox.Input> component available.

Can we not clarify this directly in the Shadcn-svelte docs???
